src/Raven.Server/Documents/ETL/Providers/Queue/QueueBrokerConnectionHelper.cs

```diff
@@ -143,12 +143,38 @@ public static QueueServiceClient CreateAzureQueueStorageServiceClient(
                     azureQueueStorageConnectionSettings.EntraId.ClientId,
                     azureQueueStorageConnectionSettings.EntraId.ClientSecret));
         }
-        else if(azureQueueStorageConnectionSettings.Passwordless != null)
+        else if (azureQueueStorageConnectionSettings.Passwordless != null)
         {
             var queueUri = new Uri($"{azureQueueStorageConnectionSettings.GetStorageUrl()}");
             queueServiceClient = new QueueServiceClient(queueUri, new DefaultAzureCredential());
         }
 
         return queueServiceClient;
     }
+
+    public static IAmazonSQS CreateAwsSqsClient(AwsSqsConnectionSettings connectionSettings)
+    {
+        AmazonSQSClient sqsClient = null;
+
+        if (connectionSettings.Basic != null)
+        {
+            var region = RegionEndpoint.GetBySystemName(connectionSettings.Basic.RegionName);
+            sqsClient = new AmazonSQSClient(connectionSettings.Basic.AccessKey, connectionSettings.Basic.SecretKey,
+                region);
+        }
+        else if (connectionSettings.Passwordless)
+        {
+            sqsClient = new AmazonSQSClient();
+        }
+        else if (connectionSettings.UseEmulator)
+        {
+            sqsClient = new AmazonSQSClient(new AmazonSQSConfig
+            {
+                ServiceURL = Environment.GetEnvironmentVariable("RAVEN_AWS_SQS_EMULATOR_URL"), 
```

Comment:
What if the env variable isn't setup? Shouldn't we throw when?

Reply:
It should be `RAVEN_AMAZON_SQS_EMULATOR_URL`

Reply:
fixed